kyo-scheduler/jvm/src/main/scala/kyo/scheduler/InternalClock.scala

```diff
@@ -0,0 +1,33 @@
+package kyo.scheduler
+
+import java.util.concurrent.Executor
+import java.util.concurrent.locks.LockSupport
+
+final private class InternalClock(executor: Executor):
```

Comment:
`Task`'s runtime tracking has to measure execution times and obtaining the time from the system clock is prohibitively expensive. This internal clock provides approximate time measurements by updating a volatile field every ~1ms. Readers only pay the price of the read barrier and eventual cache misses when the field is updated.